frontend/packages/jobs/src/prompts/generateDocsSuggestion/generateDocsSuggestion.ts

```diff
@@ -35,74 +33,258 @@ migrationPatterns.md:
 migrationOpsContext.md:
 - Operational constraints on executing migrations
 - Timing, tooling, deployment risks, safety strategies
+`
+
+// Convert schemas to JSON format for LLM
+const evaluationJsonSchema = toJsonSchema(evaluationSchema)
+const docsSuggestionJsonSchema = toJsonSchema(docsSuggestionSchema)
+
+// Common evaluation response structure for a single file
+const fileEvaluationExample = {
+  updateNeeded: 'true/false',
+  reasoning: 'Detailed explanation of why this file needs to be updated or not',
+  suggestedChanges:
+    'If updates are needed, provide specific suggestions for what should be added or modified',
+}
+
+// Example evaluation response structure for the prompt
+const evaluationResponseExample = {
+  schemaPatterns: { ...fileEvaluationExample },
+  schemaContext: { ...fileEvaluationExample },
+  migrationPatterns: { ...fileEvaluationExample },
+  migrationOpsContext: { ...fileEvaluationExample },
+}
+
+// Example update response structure for the prompt
+const updateResponseExample = {
+  schemaPatterns: 'Full updated content for schemaPatterns.md',
+  migrationPatterns: 'Full updated content for migrationPatterns.md',
+}
 
-.liamrules:
-- Informal but recurring knowledge
-- Field/model specific patterns
-- One-time decisions that may inform future work
+// Step 1: Evaluation template to determine which files need updates
+const EVALUATION_TEMPLATE = ChatPromptTemplate.fromTemplate(`
+You are Liam, an expert in schema design and migration strategy for this project.
+
+## Your Task
+Analyze the migration review and determine which documentation files need to be updated.
+
+## 📁 Documentation Structure
+${DOCS_STRUCTURE_DESCRIPTION}
 
 ---
 
-Migration Review:
+## Migration Review
+
+<text>
+
 {reviewResult}
 
-Current Documentation:
-{docsArray}
+</text>
+
+## Current Documentation
+
+<docs>
+
+{formattedDocsContent}
+
+</docs>
+
+---
+
+## Your Task
```

Comment:
:eyes: Noticed the second `## Your Task.`  That said, it’s clear and easy to understand, so I think it’s totally fine.